pkg/resource/model_package_group/hooks.go

```diff
@@ -0,0 +1,122 @@
+	// ModelPackageGroup
+	TerminalStatuses = []v1alpha1.ModelPackageGroupStatus_SDK{
+		v1alpha1.ModelPackageGroupStatus_SDK_Deleting,
+	}
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		ErrModelPackageGroupDeleting,
+		5*time.Second,
+	)
+	requeueWaitWhileInProgress = ackrequeue.NeededAfter(
+		ErrModelPackageGroupInProgress,
+		5*time.Second,
+	)
+	requeueWaitWhilePending = ackrequeue.NeededAfter(
+		ErrModelPackageGroupPending,
+		5*time.Second,
+	)
+	requeueWaitWhileDeleteFailed = ackrequeue.NeededAfter(
+		ErrModelPackageGroupDeleteFailed,
+		5*time.Second,
+	)
+)
+
+// modelPackageGroupHasTerminalStatus returns whether the supplied SageMaker ModelPackageGroup is in a
+// terminal state
+func modelPackageGroupHasTerminalStatus(r *resource) bool {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return false
+	}
+	ts := *r.ko.Status.ModelPackageGroupStatus
+	for _, s := range TerminalStatuses {
+		if ts == string(s) {
+			return true
+		}
+	}
+	return false
+}
+
+// isModelPackageGroupPending returns true if the supplied SageMaker ModelPackageGroup is in the process
+// of pending
+func isModelPackageGroupPending(r *resource) bool {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return false
+	}
+	sagemaker_status := *r.ko.Status.ModelPackageGroupStatus
+	return sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_Pending)
+}
+
+// isModelPackageGroupProgressreturns true if the supplied SageMaker ModelPackageGroup is in progress
+func isModelPackageGroupInProgress(r *resource) bool {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return false
+	}
+	sagemaker_status := *r.ko.Status.ModelPackageGroupStatus
+	return sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_InProgress)
+}
+
+// isModelPackageGroupDeleting returns true if the supplied SageMaker ModelPackageGroup is in the process
+// of being deleted
+func isModelPackageGroupDeleting(r *resource) bool {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return false
+	}
+	sagemaker_status := *r.ko.Status.ModelPackageGroupStatus
+	return sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_Deleting)
+}
+
+// isModelPackageGroupDeleting returns true if the supplied SageMaker ModelPackageGroup delete failed
+func isModelPackageGroupDeleteFailed(r *resource) bool {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return false
+	}
+	sagemaker_status := *r.ko.Status.ModelPackageGroupStatus
+	return sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_DeleteFailed)
+}
+
+func ModelPackageGroupCustomSetOutput(r *resource) {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return
+	}
+	sagemaker_status := *r.ko.Status.ModelPackageGroupStatus
+	if sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_Completed) || sagemaker_status == string(v1alpha1.ModelPackageGroupStatus_SDK_Failed) {
+		setSyncedCondition(r, corev1.ConditionTrue, nil, nil)
```

Comment:
lets pass message atleast

`ModelPackageGroup is in + *r.ko.Status.ModelPackageGroupStatus + status`